packages/meditrak-server/src/routes/dashboardRelations/assertDashboardRelationsPermissions.js

```diff
@@ -0,0 +1,282 @@
+      -- with the user's list of countries for the appropriate permission group (i.e., they have
+      -- access to at least one country within the project)
+      OR (
+        ARRAY(
+          SELECT DISTINCT e3.country_code
+          FROM entity e3
+          INNER JOIN entity_relation er3
+            ON e3.id = er3.child_id
+          INNER JOIN project p3
+            ON  er3.parent_id = p3.entity_id
+            AND er3.entity_hierarchy_id = p3.entity_hierarchy_id
+          INNER JOIN dashboard d3
+            ON p3.code = d3.root_entity_code
+          INNER JOIN dashboard_relation dr3 
+            ON dr3.dashboard_id = d3.id 
+          WHERE dr3.id = "dashboard_relation".id
+        )::TEXT[]
+        &&
+        ARRAY(
+          SELECT DISTINCT TRIM('"' FROM JSON_ARRAY_ELEMENTS(?::JSON->dr4.permission_group)::TEXT) 
+          FROM (SELECT unnest(permission_groups) AS permission_group 
+                FROM dashboard_relation subdr2
+                WHERE subdr2.id = "dashboard_relation".id) dr4
+        )
+      )
+    )
+  )`,
+    parameters: [
+      JSON.stringify(countryCodesByPermissionGroup),
+      JSON.stringify(countryCodesByPermissionGroup),
+    ],
+  };
+  return dbConditions;
+};
+
+export const createDashboardRelationsViaParentDashboardDBFilter = (
+  accessPolicy,
+  criteria,
+  options,
+  dashboardId,
+) => {
+  if (hasBESAdminAccess(accessPolicy)) {
+    return { dbConditions: criteria, dbOptions: options };
+  }
+
+  const dbConditions = { ...criteria };
+  const allPermissionGroups = accessPolicy.getPermissionGroups();
+  const countryCodesByPermissionGroup = {};
+
+  // Generate lists of country codes we have access to per permission group
+  allPermissionGroups.forEach(pg => {
+    countryCodesByPermissionGroup[pg] = accessPolicy.getEntitiesAllowed(pg);
+  });
+
+  const dbOptions = {
+    ...options,
+    sort: ['dashboard_relation.sort_order'],
+  };
+  dbOptions.multiJoin = mergeMultiJoin(
+    [
+      {
+        joinWith: TYPES.DASHBOARD,
+        joinCondition: ['dashboard_relation.dashboard_id', 'dashboard.id'],
+      },
+    ],
+    dbOptions.multiJoin,
+  );
+
+  dbConditions[RAW] = {
+    sql: `
+      dashboard_id = ?
+      AND 
+      (
+        dashboard.root_entity_code = ANY
+        (ARRAY(
+          SELECT DISTINCT TRIM('"' FROM JSON_ARRAY_ELEMENTS(?::JSON->dr2.permission_group)::TEXT) 
+          FROM (SELECT unnest(permission_groups) AS permission_group 
+                FROM dashboard_relation subdr1
+                WHERE subdr1.id = "dashboard_relation".id) dr2
+        ))
+        -- for projects, pull the country codes from the child entities and check that there is overlap
+        -- with the user's list of countries for the appropriate permission group (i.e., they have
+        -- access to at least one country within the project)
+        OR (
+          ARRAY(
+            SELECT DISTINCT e3.country_code
```

Comment:
Ahh it's because in the above sub query, I already use suffix `2`. My justification is I just want to have increment suffix for each sub query so that it is easier to track

Reply:
Makes sense! I read it as "reusing the same table" which wasn't the case here, so I got a bit confused